src/Core/Framework/Analytics/TokenService.php

```diff
@@ -0,0 +1,108 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Framework\Analytics;
+
+use Psr\Log\LoggerInterface;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\System\SystemConfig\SystemConfigService;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+#[Package('data-services')]
+class TokenService
+{
+    private const CONFIG_KEY_ANALYTICS_SECRET = 'core.analytics.secret';
+
+    private const RESPONSE_KEY_TOKEN = 'token';
+    private const RESPONSE_KEY_EXPIRES_AT = 'expires_at';
+
+    /**
+     * @internal
+     */
+    public function __construct(
+        private readonly SystemConfigService $systemConfigService,
+        private readonly HttpClientInterface $client,
+        private readonly LoggerInterface $logger,
+        private readonly string $gatewayBaseUrl
+    ) {
+    }
+
+    public function generate(string $referer): ?Token
+    {
+        $secret = $this->systemConfigService->get(self::CONFIG_KEY_ANALYTICS_SECRET);
```

Comment:
How "secret" is this supposed to be? The system config is widely accessible over API by default